### ISSUE TITLE
"Point" style coding with fp-ts

### ISSUE THREAD

**Issue by ChuckJonas** [ChuckJonas/mostly-adequate-fp-ts#14]
Starting with CH08, the book often introduces concepts in the "point" style:

```js
Maybe.of(add).ap(Maybe.of(2)).ap(Maybe.of(3));
// Maybe(5)

Task.of(add).ap(Task.of(2)).ap(Task.of(3));
// Task(5)
```

AFAIK, `fp-ts` doesn't really provide the ability to use functors in this way (maybe I'm wrong)?

It would be good to add a note on this, along side an example translation.

**Comment by cdimitroulas** [comment]
>AFAIK, fp-ts doesn't really provide the ability to use functors in this way (maybe I'm wrong)?

That's correct. It actually used to support this chainable style but it moved to the `pipe` with `fp-ts` version 2.0. You can read more about it in this issue if you're curious https://github.com/gcanti/fp-ts/issues/823 (warning: it's a super long discussion :joy:)

**Comment by ChuckJonas** [comment]
nice, maybe we add a note and just link the issue for anyone who is interested 👍

**Comment by ChuckJonas** [comment]
Is there an equaivent type to the container:

```ts
// (a -> b) -> Container a -> Container b
Container.prototype.map = function (f) {
  return Container.of(f(this.$value));
};

Container.of(2).map(two => two + 2); 
// Container(4)

Container.of('flamethrowers').map(s => s.toUpperCase()); 
// Container('FLAMETHROWERS')

Container.of('bombs').map(append(' away')).map(prop('length')); 
// Container(10)
```

I'm guessing not as it only really makes sense if you are chaining/mapping in the point style

**Comment by cdimitroulas** [comment]
The `Identity` functor/monad could be similar to `Container` - https://gcanti.github.io/fp-ts/modules/Identity.ts.html

Not sure if that's more confusing than helpful though
